tests/conftest.py

```diff
@@ -0,0 +1,186 @@
+                warnings.simplefilter('error')
+            elif self.type == "csvt:NegativeJsonTest":
+                stack.enter_context(warnings.catch_warnings())
+                warnings.simplefilter('ignore')
+                stack.enter_context(pytest.raises(ValueError))
+
+            elif self.type == "csvt:PositiveValidationTest":
+                # Turn warnings into exceptions!
+                stack.enter_context(warnings.catch_warnings())
+                warnings.simplefilter('error')
+            elif self.type == "csvt:NegativeValidationTest":
+                # Warnings count as negative validatio, too!
+                stack.enter_context(pytest.raises(ValueError))
+                #stack.enter_context(warnings.catch_warnings())
+                #warnings.simplefilter('error')
+            elif self.type == "csvt:WarningValidationTest":
+                stack.enter_context(pytest.warns(UserWarning))
+
+            ds = CSVW(
+                self.action, md_url=self.option.get('metadata'), validate=self.is_validation_test)
+            if self.is_validation_test:
+                if self.type == 'csvt:PositiveValidationTest':
+                    assert ds.is_valid
+                elif self.type == 'csvt:WarningValidationTest':
+                    if not ds.is_valid:
+                        warnings.warn('invalid')
+                else:
+                    if not ds.is_valid:
+                        raise ValueError('invalid')
+
+            elif self.is_json_test:
+                assert unorder(ds.to_json(minimal=self.option.get('minimal'))) == \
+                       unorder(get_json(self.result)), \
+                    '{}: {}'.format(self.id, self.name)
+
+    def run(self):
+        import requests_mock
+
+        def text_callback(request, context):
+            url = urllib.parse.urlparse(request.url)
+            if url.netloc == 'www.w3.org':
+                if url.path.startswith('/2013/csvw/tests/'):
+                    p = csvw_tests_path(url.path.replace('/2013/csvw/tests/', ''))
+                    if p.exists():
+                        context.status_code = 200
+                        return p.read_text(encoding='utf8')
+                elif url.path == '/.well-known/csvm':
+                    context.status_code = 200
+                    return """{+url}-metadata.json
+csv-metadata.json
+{+url}.json
+csvm.json
+"""
+                context.status_code = 404
+                return ''
+            raise ValueError(request.url)  # pragma: no cover
+
+        with requests_mock.Mocker() as mock:
+            if self.contentType:
+                mock.head(self.action, text='', headers={'Content-Type': self.contentType})
+            elif self.httpLink:
+                mock.head(self.action, text='', headers={'Link': self.httpLink})
+            else:
+                mock.head(self.action, text='', headers={})
+            mock.get(requests_mock.ANY, text=text_callback)
```

Comment:
`requests-mock` is a pretty cool way to access the files in the w3c/csvw submodule "as if it were over HTTP".